src/components/new-safe/create/steps/ReviewStep/index.tsx

```diff
@@ -263,7 +263,7 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
       gtmSetChainId(chain.chainId)
 
       if (isCounterfactualEnabled && payMethod === PayMethod.PayLater) {
-        router?.push({
+        await router?.push({
```

Comment:
I thought I added this already in the past but this prevents the success screen from being shown until the dashboard is opened otherwise it pops up immediately even while still seeing the review screen in the background